src/planwise/client/asdf.cljs

```diff
@@ -60,3 +60,8 @@
 (defn reload!
   [asdf]
   (assoc asdf :state :reloading))
+
+(defn update
+  "Updates the value in the asdf preserving its current state"
+  [asdf update-fn & args]
+  (assoc asdf :value (apply update-fn (:value asdf) args)))
```

Comment:
Well, not exactly since this doesn't reset the state. But this shouldn't be needed. See my other comment.